BedrockCore.cpp

```diff
@@ -64,6 +64,70 @@ bool BedrockCore::isTimedOut(unique_ptr<BedrockCommand>& command) {
     return false;
 }
 
+void BedrockCore::prePeekCommand(unique_ptr<BedrockCommand>& command) {
+    AutoTimer timer(command, BedrockCommand::PREPEEK);
+    BedrockServer::ScopedStateSnapshot snapshot(_server);
+    command->lastPeekedOrProcessedInState = _server.getState();
+
+    // Convenience references to commonly used properties.
+    const SData& request = command->request;
+    SData& response = command->response;
+    STable& content = command->jsonContent;
+
+    try {
+        SDEBUG("prePeeking at '" << request.methodLine << "' with priority: " << command->priority);
+        uint64_t timeout = _getRemainingTime(command, false);
+        command->prePeekCount++;
+
+        _db.startTiming(timeout);
+
+        if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
+            STHROW("501 Failed to begin shared prePeek transaction");
+        }
+
+        // Make sure no writes happen while in prePeek command
+        _db.setQueryOnly(true);
+
+        // prePeek.
+        command->reset(BedrockCommand::STAGE::PREPEEK);
+        command->prePeek(_db);
+        SDEBUG("Plugin '" << command->getName() << "' prePeeked command '" << request.methodLine << "'");
+
+        if (command->httpsRequests.size()) {
+            STHROW("405 https requests cannot be made in prePeek");
```

Comment:
This is a weird error here. Why do we have this?

Reply:
This prevents anyone from trying to make an https request in prePeek.

Reply:
I think it's weird to do this after the fact. We also have never enforced this anywhere else, so it's sort of weird to add here. For instance, we don't check it in `postProcess`. I think if we want to check this, we should do it here:
https://github.com/Expensify/Bedrock/blob/main/libstuff/SHTTPSManager.cpp#L23

And check that we're in `peek` specifically.

Reply:
Discussed 1:1